compiler/damlc/tests/daml-test-files/ConstrainedClassMethod.daml

```diff
@@ -0,0 +1,13 @@
+{-# LANGUAGE ConstrainedClassMethods #-}
+
+daml 1.2
+-- | This module tests the case where a class method contains a constraint
+-- not present in the class itself.
+module ConstrainedClassMethod where
+
+class A t where
```

Comment:
fwiw it’s not that much of an edge case `foldMap` is the first non-artifical example that comes to mind.

Reply:
Good point! 👍